static/js/stream_edit.js

```diff
@@ -700,7 +708,13 @@ export function initialize() {
             open_edit_panel_for_row($stream_row);
         }
         stream_ui_updates.update_regular_sub_settings(sub);
-
+        // Refresh the view only if the stream is unsubscribed before
+        // and subscribed now
+        if (!sub.subscribed) {
+            narrow.activate([{operator: "stream", operand: sub.name}], {
+                trigger: "subscribe button",
```

Comment:
We should narrow to the stream (or topic if required) only if the user was narrowed to the stream to which the user subscribed.

Here also, we should narrow to topic if user was narrowed to topic before subscribing.